File: ch5/randomize.java
import java.util.Random;
import java.util.Arrays;
public class ShuffleRand
{ 
    static void randomize( int arr[])
    { 
        int n = arr.length;
        Random r = new Random();   
        for (int i = n-1; i > 0; i--) { 
            int j = r.nextInt(i+1); 
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        } 
        System.out.println(Arrays.toString(arr));
    } 
    public static void main(String[] args)
    {
         
        int[] arr = {1, 2, 3, 4, 5, 6, 7}; 
        randomize (arr);
    }
} 

File: ch5/randomize.js
const print = arr => {
  let ret = ""
  for (let a of arr) {
    ret += a + ' '
  }
  console.log(ret)
}
const randomize = (arr) => {
  const n = arr.length
  for (let i = n - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [arr[i], arr[j]] = [arr[j], arr[i]]; 
  }
}
let arr = [1, 2, 3, 4, 5, 6, 7];
randomize(arr);
print(arr) 